### PR TITLE
[Plugin] showBadgesInChat

### PR DESCRIPTION
**Pull request by ShalevG114**
Allows the user to see Vencord and profile badges in the chat.

### CODE REVIEW

File: src/plugins/showBadgesInChat/index.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { findByProps } from "@webpack";
+import { useEffect, useState } from "@webpack/common";
+
+let BadgeComponent: React.ComponentType<any> = () => null;
+let badgeClassName: string = "";
+
+const dcBadges: [number, string, string][] = [
```
Comment (MrDiamondDog) [inline src/plugins/showBadgesInChat/index.tsx:17]: Is there a reason for using an array over an object?

**Review by MrDiamondDog — CHANGES_REQUESTED** [review]

File: src/plugins/showBadgesInChat/index.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    [18, "Moderator Programs Alumni", "fee1624003e2fee35cb398e125dc479b"]
+];
+Object.freeze(dcBadges);
+
+function getUserDCBadges(flags: number): [string, string][] {
+    const userDCBadges: [string, string][] = [];
+    for (let i: number = 0; i < dcBadges.length; i++)
+        if (flags & (1 << dcBadges[i][0]))
+            userDCBadges.push([dcBadges[i][1], dcBadges[i][2]]);
+    return userDCBadges;
```
Comment (FieryFlames) [inline src/plugins/showBadgesInChat/index.tsx:38]:
```suggestion
    return dcBadges
      .filter(badge => flags & (1 << badge[0]))
      .map(badge => [badge[1], badge[2]])
```

File: src/plugins/showBadgesInChat/index.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+function vencordContributorChatBadge(userID: string) {
+    return isPluginDev(userID) ? [
+        <span style={{ order: settings.store.vencordContributorBadgePosition }}>
+            <RoleIconComponent
+                className={roleIconClassName}
+                name={"Vencord Contributor"}
+                size={20}
+                src={"https://cdn.discordapp.com/attachments/1033680203433660458/1092089947126780035/favicon.png"}
```
Comment (KrstlSkll69) [inline src/plugins/showBadgesInChat/index.tsx:74]:
```suggestion
                src={"https://vencord.dev/assets/favicon.png"}
```

Comment (KrstlSkll69) [reply src/plugins/showBadgesInChat/index.tsx:74]: possible fix for the dead link

File: src/plugins/showBadgesInChat/index.tsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+ */
+
+import { addDecoration, removeDecoration } from "@api/MessageDecorations";
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import { isPluginDev } from "@utils/misc";
+import definePlugin, { OptionType } from "@utils/types";
+import { findByProps } from "@webpack";
+import { useEffect, useState } from "@webpack/common";
+
```
Comment (KrstlSkll69) [inline src/plugins/showBadgesInChat/index.tsx:14]:
```suggestion
let VencordBadges = {} as Record<string, Array<Record<"tooltip" | "badge", string>>>;
```

**Comment by Vendicated** [comment]
superseded by #2347, thanks for your work!

**Pull request closed without merging**